doc/ext/traffic-server.py

```diff
@@ -371,6 +371,26 @@ def xref_cleanup(app, env, node, contnode):
                 return node
     return;
 
+def make_github_link(name, rawtext, text, lineno, inliner, options={}, content=[]):
+    """
+    This docutils role lets us link to source code via the handy :ts:git: markup.
+    Link references are rooted at the top level source directory. All links resolve
+    to GitHub.
+
+    Examples:
+
+        To link to proxy/Main.cc:
+
+            Hi, here is a link to the proxy entry point: :ts:git:`proxy/Main.cc`.
+
+        To link to CONTRIBUTING.md:
+
+            If you want to contribute, take a look at :ts:git:`CONTRIBUTING.md`.
+    """
+    url = 'https://github.com/apache/trafficserver/blob/master/{}'
```

Comment:
That's a valid point. What if we pinned the references to the most recent commit of the referenced document? That way we can even add in line numbers.

The downside to that is that outdated documentation may be talking about stuff that doesn't exist anymore. But in a perfect world that shouldn't happen because everyone is updating the documentation as they change code :).